homework_4/Barbhaiwala/solution.js

```diff
@@ -0,0 +1,73 @@
+var colors = require("colors");
+
+function HotelParser(jsonString) {
+    
+    this.hotelInfoJson = jsonString; 
+    
+    this.parse = function () { 
+        var parsedHotelInfo = parseHotelInfoFromJson();
+        
+        var hotelList = verify(parsedHotelInfo.hotelList);
+        return hotelList;
+    }
+    
+    function verify(hotelList) {
```

Comment:
Needs a better name since it also returns an array of `HotelInformation` objects. Single responsibility principle says you should make a different function for the `HotelInformation` list creation.